core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -215,14 +227,112 @@ public boolean isAvailable() {
       return file.exists();
    }
 
+   private boolean isCorruptData() throws IOException {
+      ByteBuffer buf = ByteBuffer.allocate(KEY_POS_LATEST);
+      long pos = filePos;
+      buf = readChannel(buf, pos, KEY_POS_LATEST);
+      if (buf.remaining() > 0)
```

Comment:
can someone explaining the `buf.remaining() > 0` conditions? shouldn't the `remaining()` be `0` if the `FileEntry` is correct?

Reply:
I think I got it now :) `buf.remaining() > 0` means nothing was read (i.e. the store is empty).
Maybe add a comment or I will forget again :rofl: